privacyidea/api/healthcheck.py

```diff
@@ -0,0 +1,103 @@
+from flask import Blueprint, current_app
+
+from privacyidea.api.lib.utils import send_result
+from privacyidea.lib.crypto import get_hsm
+from privacyidea.lib.resolver import get_resolver_list, get_resolver_class
+import logging
+
+log = logging.getLogger(__name__)
+
+healthz_blueprint = Blueprint('healthz_blueprint', __name__)
+
+
+@healthz_blueprint.route('/', methods=['GET'])
+def healthz():
+    """
+    Health check endpoint that indicates if the app is healthy (running and ready to serve requests).
+    """
+    app = current_app._get_current_object()
+    if not app.config.get('APP_READY'):
+        return send_result({"status": "not healthy"}), 503
+    elif not get_hsm().is_ready:
+        return send_result({"status": "not healthy"}), 503
```

Comment:
can we return some additional information here, that the app is not healthy due to the HSM? Or is that not intended?

Reply:
fixed with commit 680e7655f0874d02c883a1fdcb4dc6647c3104df